### ISSUE TITLE
clang: error: no such file or directory: '...libOpenTokReactNative.a'

### ISSUE THREAD

**Issue by TeruyaHaroldo** [opentok/opentok-react-native#123]
Remember this: **Can not build in physical device iPhone 5s but can build in the emulator.**

I did the following instructions in sequence:

``react-native init test``

Changed in the ``test``'s ``package.json`` the react-native version to ``0.51.0`` and did the ``yarn install``.

The ``package.json``:

```json
{
  "name": "test",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start"
  },
  "dependencies": {
    "react": "16.3.1",
    "react-native": "0.51.0"
  }
}
```

Did: ``npm install opentok-react-native``

![1](https://user-images.githubusercontent.com/6924285/42064196-adb01058-7b0b-11e8-8a00-b7b09c111723.png)

Did: ``cd ios `` then  ``pod init``

Added in the Podfile: ``pod 'OpenTok'``

```json
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'test' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for test
  pod 'OpenTok'

  target 'test-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

  target 'testTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

target 'test-tvOS' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for test-tvOS

  target 'test-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

Did ``pod install``:

![2](https://user-images.githubusercontent.com/6924285/42064275-06cd9296-7b0c-11e8-8e7c-aa484ff4b103.png)

Removed the conflicts: 

```json
target 'test-tvOSTests' do
  inherit! :search_paths
  # Pods for testing
end
```

And the Podfile became like this:

```json
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'test' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for test
  pod 'OpenTok'

  target 'testTests' do
    inherit! :search_paths
    # Pods for testing
  end

end

target 'test-tvOS' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for test-tvOS

  target 'test-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

Did ``pod install``:

![3](https://user-images.githubusercontent.com/6924285/42064296-1b3d25e8-7b0c-11e8-96e8-4b590e763987.png)

Did: ``yarn install`` and ``react-native link opentok-react-native``:

![4](https://user-images.githubusercontent.com/6924285/42064313-2726b770-7b0c-11e8-8e14-46b6ec14acdc.png)

The ios folder:

<img width="203" alt="5" src="https://user-images.githubusercontent.com/6924285/42064316-29a00fd8-7b0c-11e8-8241-e61cb74ac6f3.png">

Created ``OTInstall`` and the auto-generated ``test-Bridging-Header.h``:

![6](https://user-images.githubusercontent.com/6924285/42064321-2fb3cc70-7b0c-11e8-82ab-9811dcada730.png)

Trying to compile:

![7](https://user-images.githubusercontent.com/6924285/42064324-3226d1c8-7b0c-11e8-94bb-2a4840d17df0.png)

Fixed:

![8](https://user-images.githubusercontent.com/6924285/42064333-347c2ae0-7b0c-11e8-935f-46cb70f3c147.png)

And now

![9](https://user-images.githubusercontent.com/6924285/42064336-373cd720-7b0c-11e8-8aa7-8ac92fba1c7a.png)

Re-added the ``OpenTokReactNative.xcodeproj`` project from the node_modules and the ``libOpenTokReactNative.a`` in the 'Link Binary With Libraries' in the 'Target':

![10](https://user-images.githubusercontent.com/6924285/42064340-3ca85a9a-7b0c-11e8-9fc7-3f58abe2f830.png)

And then:

![11](https://user-images.githubusercontent.com/6924285/42064351-477b8dfc-7b0c-11e8-999f-4b15511c6dea.png)

Tried many thinks like ``clean``, deleted ``DerivedData``, ``yarn upgrade opentok-react-native``, ``react-native run-ios --device <name of the iphone>``, deleting and adding many and many times the ``node_modules``. Removed and added many time the ``OpenTokReactNative.xcodeproj``in the ``test`` project ``Libraries``.

One solution that I found is to copy and paste the files of the ``OpenTokReactNative``  into the ``test`` project:

![12](https://user-images.githubusercontent.com/6924285/42064832-3576fd88-7b0e-11e8-8d85-be4d2f9b9798.png)

In this situation, I can compile and use ``OpenTok``. I had to remove the ``OpenTokReactNative.xcodeproj`` from the ``Libraries``.

Remember this: **Can not build in the physical device iPhone 5s but can build in the emulator.**

In another much bigger project, the error is different, but the solution is the same:

![captura de tela 2018-06-28 as 14 24 03](https://user-images.githubusercontent.com/6924285/42065073-459e6880-7b0f-11e8-9c1b-3e6351f74c5f.png)

Remember this: **Can not build in the physical device iPhone 5s but can build in the emulator.**

**Comment by HaroldoTeruya** [comment]
Updating.... In the emulator is not running too. Help!

**Comment by msach22** [comment]
@HaroldoTeruya Thanks for sharing your findings. Could you please share which `iOS` version the 5s device and similar are running? Are they running a minimum of iOS 9?

**Comment by HaroldoTeruya** [comment]
On Device iOS 10.3.3 is not running.
On emulator in any version and device is running.

And i tested they running on a minimum of iOS 9.

**Comment by msach22** [comment]
@HaroldoTeruya Sorry for this blocker. I'm happy to set up a 30 minute chat and go over the issue. Can you send me an email at manik@tokbox.com so we can schedule a time?

**Comment by msach22** [comment]
@HaroldoTeruya Just pinging you for an update

**Comment by renanpupin** [comment]
Opentok not compiling in iPhone 5s (device and simulator).

**Comment by riadhriadh** [comment]
i thenk  you hed manual delet  lib

**Comment by msach22** [comment]
Hi folks, the next release, `0.9.0`, should be fix this because the target is being set to 9.0

**Comment by msach22** [comment]
This has been fixed and released of `v0.9.9`
